### DISCUSSION TITLE
氣象資料開放平臺API-用Python查地震報告

### DISCUSSION THREAD

**Discussion by gnsJhenJie** [1101-GS4719-C/Python-Community-GS4719-C#510] [General]
今天發生了好多地震，想用Python來查地震報告嗎? 中央氣象局提供了Open API的平台，讓大家可以輕鬆地透過API抓取各類氣象資料，當然地震報告也包括在內， 那我們現在就來看看如何透過Python抓取地震資料吧！
## 註冊氣象資料開放平臺會員
先到[https://opendata.cwb.gov.tw/userLogin](https://opendata.cwb.gov.tw/userLogin) 註冊會員(Facebook登入目前無法使用)，接著到[API授權碼](https://opendata.cwb.gov.tw/user/authkey)頁面產生並記下屬於自己的授權碼(**API授權碼不要讓別人知道喔**)
<img width="1486" alt="Screen Shot 2022-03-23 at 14 45 39" src="https://user-images.githubusercontent.com/32983347/159639374-5f2e2835-fbb1-4e50-8be8-358259b2317f.png">
有了授權碼後就可以使用氣象局提供的各個API了。
## 找尋適合的API
接著需要到[說明文件](https://opendata.cwb.gov.tw/dist/opendata-swagger.html#/)尋找適合使用的API，經過一番尋找後...看起來這支API就是我們要的了。  
<img width="1486" alt="Screen Shot 2022-03-23 at 14 50 24" src="https://user-images.githubusercontent.com/32983347/159640087-71774e9d-13fa-41ce-b866-0d5efb2cdd19.png">

可以先在[說明文件](https://opendata.cwb.gov.tw/dist/opendata-swagger.html#/%E5%9C%B0%E9%9C%87%E6%B5%B7%E5%98%AF/get_v1_rest_datastore_E_A0015_001)的網頁試試看這支API會回傳的資料內容
<img width="1486" alt="Screen Shot 2022-03-23 at 14 51 47" src="https://user-images.githubusercontent.com/32983347/159640282-7b269681-9353-473b-a017-7fcdc43638b5.png">

## 透過Python抓取API資料
接著就是透過Python 來取得API的資料啦~ 程式如下，大家可以自己更改內容取得不同的資料！
```python
import requests
import json
token = "CWB-E96B2DF1-95E9-42E7-978D-99154DD54E22" # 換成自己的API授權碼
limit = 20 # 取得最近20筆地震報告
req = requests.get(
    f"https://opendata.cwb.gov.tw/api/v1/rest/datastore/E-A0015-001?Authorization={token}&limit={limit}") # API的接入點
data = json.loads(req.text)
earthquakes = data["records"]["earthquake"]
for earthquake in earthquakes:
    print("地震編號:", earthquake["earthquakeNo"])
    for area in earthquake["intensity"]["shakingArea"]:
        print(area["areaName"], area["areaIntensity"]
              ["value"], area["areaIntensity"]["unit"])
```
Output:
<img width="513" alt="Screen Shot 2022-03-23 at 14 57 03" src="https://user-images.githubusercontent.com/32983347/159641046-a2ce4653-0690-4ecd-9ff3-8c6f9d670dc8.png">

**Comment by ghost** [comment]
看到短時間就能使用API跟時事結合，肯定是個霸...

**Comment by minniewei** [comment]
非常感謝大佬的提供！！！未來如果做跟地震相關的題材的話能夠有效利用

**Comment by ThomasMeng0515** [comment]
懂得網路爬蟲真的會是一件很方便的事，可惜我不會QQ

**Comment by JunTingLin** [comment]
據我所知有些api的是需要token等授權碼，甚至比如說[google API](https://cloud.google.com/apis/)還要收取相應的費用
![image](https://user-images.githubusercontent.com/92431095/159732990-8796b6f6-024c-43ad-8ed6-af01eb6ec924.png)

但大多的api都是json開源且免費的，比如[YouBike2.0臺北市公共自行車即時資訊](https://data.gov.tw/dataset/137993)
![image](https://user-images.githubusercontent.com/92431095/159733508-3fe43000-86f7-4172-a08f-ec5a817fcb67.png)

[政府資料開放平台
](https://data.gov.tw/)就是開源資料的集大成
![logo 4fd0012](https://user-images.githubusercontent.com/92431095/159734556-08735773-93ea-4d08-83e5-9e5fe5432767.svg)

**Comment by chia-kai-chang** [comment]
同學把時事跟Python結合，很棒唷~ 讚讚~

**Comment by HuanYuuuuuu** [comment]
欸好強，將身邊的資源做了非常好的利用。希望我改天也可以跟你一樣

**Comment by Lai27330** [comment]
感覺是一個很有效率的地震資訊收集方法，能精準的找到需要的資訊而不被其他雜亂的數據影響閱讀的方便度，好厲害

**Comment by gnsJhenJie** [comment]
https://opendata.cwb.gov.tw/dist/opendata-swagger.html#/ 上還有很多可以玩的資料，像是天氣預報也是很實用的資料

**Comment by dayu1004** [comment]
太酷了吧 我從來沒想過地震會跟 python 有關係 這篇分享又讓我再次感嘆科技的強大了

**Comment by hapiwangy** [comment]
有些時候透過資料的收集和分析可以讓我們對於災害處裡和防護上能有幫助。

**Comment by PoooYen** [comment]
太用心了! 真的很實用呢!

**Comment by Sazmafuji** [comment]
原來還有這種結合方式，想必此會在預測地震上有很大幫助

**Comment by sean1325418** [comment]
用這例子來學從網路抓資料滿淺顯易懂的

**Comment by xkllkx** [comment]
謝謝分享，原po太強大了...

**Comment by Yuhsin000** [comment]
這篇文章可以存起來ㄟ  ，之後複製貼上，會用就可以有大大的幫助了
